### PR TITLE
Add support for DateTime

### PR DESCRIPTION
**Pull request by joelwurtz**
See #16, only date time for the moment (but one of the most important)

poke @sagikazarmark @pyrech

### CODE REVIEW

File: src/Guesser/Guess/DateTimeType.php [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+/**
+ * Represent a DateTime type.
+ */
+class DateTimeType extends ObjectType
+{
+    private $format;
+
+    public function __construct($object, $format = DATE_RFC3339)
+    {
+        parent::__construct($object, '\DateTime', []);
```
Comment (damienalexandre) [inline src/Guesser/Guess/DateTimeType.php]: Could it support `DateTimeInterface` to also support `DateTimeImmutable`?

File: src/Guesser/Guess/DateTimeType.php [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+use PhpParser\Node\Scalar;
+
+/**
+ * Represent a DateTime type.
+ */
+class DateTimeType extends ObjectType
+{
+    private $format;
+
+    public function __construct($object, $format = DATE_RFC3339)
```
Comment (sagikazarmark) [inline src/Guesser/Guess/DateTimeType.php]: Why not use `\DateTime::RFC3339`?

File: src/Guesser/Guess/DateTimeType.php (hunk 1 above)
Comment (joelwurtz) [inline src/Guesser/Guess/DateTimeType.php]: Hmm thinking of just using DateTimeImmutable and nothing else, if someone wants a DateTime he can create it's own type, but don't to have too much generic thing, also supporting a Interface would be hard (it's change the way of generating type hint and other stuff)

File: src/Guesser/Guess/DateTimeType.php (hunk 2 above)
Comment (joelwurtz) [inline src/Guesser/Guess/DateTimeType.php]: Same thing, ide propose this one first, but can change

File: src/Guesser/Guess/DateTimeType.php (hunk 1 above)
Comment (sagikazarmark) [inline src/Guesser/Guess/DateTimeType.php]: Nah, I don't think DateTimeImmutable is more commonly used than DateTime.

Comment (joelwurtz) [inline src/Guesser/Guess/DateTimeType.php]: After thinking, i will treat DateTimeImmutable later with a global parameter when generating : `$jane->generateImmutable()`

This generator will create every method of models and generate the correct normalizer code to have a immutable.

I think it make more sense : either having nothing immutable or everything

Comment (pyrech) [inline src/Guesser/Guess/DateTimeType.php]: :+1:

**Pull request merged**